hail/src/main/scala/is/hail/expr/ir/LowerMatrixIR.scala

```diff
@@ -812,17 +813,20 @@ object LowerMatrixIR {
       case MatrixAggregate(child, query) =>
         val lc = lower(child, ab)
         val idx = Symbol(genUID())
-        lc
-          .aggregate(
-            aggLet(
-              __entries_field = 'row (entriesField),
-              __cols_field = 'global (colsField)) {
-              irRange(0, '__entries_field.len)
-                .filter(idx ~> !'__entries_field (idx).isNA)
-                .aggExplode(idx ~> aggLet(sa = '__cols_field (idx), g = '__entries_field (idx)) {
-                  subst(query, matrixSubstEnv(child))
-                })
-            })
+        TableAggregate(lc,
+          aggExplodeIR(filterIR(
```

Comment:
The indentation is making this hard to parse. Could you maybe start the `filterIR` on the next line and indent its arguments, and put the filter body on its own line?

Reply:
oh, sure.